gitpod-io/gitpod
# Buffer logs to not overflow xterm

## Description

This PR adds changes which require xterm to signal it's ready to accept new data before actually sending it. It is implemented with a very simple buffer sending (at max) 4kB chunks for xterm to process.

## Related Issue(s)

Attempts to fix ENT-502

## How to test

https://ft-lock-log-emitter.preview.gitpod-dev.com/workspaces

Run a big prebuild like one from https://github.com/filiptronicek/tests-with-spring-petclinic and while it runs, switch tasks frequently. Everything should be smooth as butter.
